parachains/runtimes/collectives/collectives-polkadot/src/fellowship/migration.rs

```diff
@@ -0,0 +1,259 @@
+			(1, hex_literal::hex!("8c232c91ef2a9983ba65c4b75bb86fcbae4d909900ea8aa06c3644ca1161db48"),),
+			(1, hex_literal::hex!("78e4813814891bd48bc745b79254a978833d41fbe0f387df93cd87eae2468926"),),
+			(1, hex_literal::hex!("d44824ac8d1edecca67639ca74d208bd2044a10e67c9677e288080191e3fec13"),),
+			(1, hex_literal::hex!("585e982d74da4f4290d20a73800cfd705cf59e1f5880aaee5506b5eaaf544f49"),),
+			(1, hex_literal::hex!("d851f44a6f0d0d2f3439a51f2f75f66f4ea1a8e6c33c32f9af75fc188afb7546"),),
+			(1, hex_literal::hex!("dca89b135d1a6aee0a498610a70eeaed056727c8a4d220da245842e540a54a74"),),
+			(1, hex_literal::hex!("aa91fc0201f26b713a018669bcd269babf25368eee2493323b1ce0190a178a27"),),
+			(1, hex_literal::hex!("dc20836f2e4b88c1858d1e3f918e7358043b4a8abcd2874e74d91d26c52eca2a"),),
+			(1, hex_literal::hex!("145d6c503d0cf97f4c7725ca773741bd02e1760bfb52e021af5a9f2de283012c"),),
+			(1, hex_literal::hex!("307183930b2264c5165f4a210a99520c5f1672b0413d57769fabc19e6866fb25"),),
+			(1, hex_literal::hex!("6201961514cf5ad87f1c4dd0c392ee28231f805f77975147bf2c33bd671b9822"),),
+			(1, hex_literal::hex!("c6f57237cd4abfbeed99171495fc784e45a9d5d2814d435de40de00991a73c06"),),
+			(1, hex_literal::hex!("c1df5c7e8ca56037450c58734326ebe34aec8f7d1928322a12164856365fea73"),),
+			(1, hex_literal::hex!("12c039004da5e1e846aae808277098c719cef1f4985aed00161a42ac4f0e002f"),),
+			(1, hex_literal::hex!("7460ac178015d2a7c289bb68ef9fdaac071596ab4425c276a0040aaac7055566"),),
+			(1, hex_literal::hex!("eec4bd650a277342ebba0954ac786df2623bd6a9d6d3e69b484482336c549f79"),),
+			(1, hex_literal::hex!("e287c7494655d636a846f5c3347ad2cb3c462a8d46e0832be70fcc0ab54ee62d"),),
+			(1, hex_literal::hex!("82bf733f44a840f0a5c1935a002d4e541d81298fad6d1da8124073485983860e"),),
+		];
+	}
+
+	/// Implements `OnRuntimeUpgrade` trait.
+	pub struct Migration<T, I = ()>(PhantomData<(T, I)>);
+
+	impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for Migration<T, I>
+	where
+		<T as frame_system::Config>::AccountId: From<[u8; 32]>,
+	{
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let onchain_version = RankedCollective::<T, I>::on_chain_storage_version();
+			assert_eq!(onchain_version, 0, "the storage version must be 0.");
+			let member_count = MemberCount::<T, I>::get(0);
+			assert_eq!(member_count, 0, "the collective must be uninitialized.");
+
+			Ok(Vec::new())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let current_version = RankedCollective::<T, I>::current_storage_version();
+			let onchain_version = RankedCollective::<T, I>::on_chain_storage_version();
+			let mut weight = T::DbWeight::get().reads(1);
+			log::info!(
+				target: TARGET,
+				"running migration with current storage version {:?} / onchain {:?}.",
+				current_version,
+				onchain_version
+			);
+			if onchain_version != 0 {
+				log::warn!(
+					target: TARGET,
+					"unsupported storage version, skipping import_fellowship migration."
+				);
+				return weight
+			}
```

Comment:
Not sure that this bring any value?

Reply:
We normally try to write migrations fault tolerant - in case that they are included twice.  
Or what do you mean?

Reply:
Yeah, but that is done here by checking if there are already members included in the pallet.